src/distilabel/llm/base.py

```diff
@@ -72,11 +72,11 @@ def __del__(self) -> None:
             self.thread_pool_executor.shutdown()
 
     def __repr__(self) -> str:
-        return f"{self.__class__.__name__}(task={self.task.__class__.__name__}, num_threads={self.thread_pool_executor._max_workers}, promp_format='{self.prompt_format}', model='{self.model_name}')"
+        return f"{self.__class__.__name__}(task={self.task.__class__.__name__}, num_threads={self.thread_pool_executor._max_workers if self.thread_pool_executor is not None else 0}, promp_format='{self.prompt_format}', model='{self.model_name}')"
```

Comment:
maybe it's better to create a property that returns the number of threads containing this `if` logic and that we can use in both methods

Reply:
@gabrielmbmb I have not a lot of experience on properties. How shall I do it on this case to avoid the if statement?

Reply:
Sorry for the intrusion. Check the following @ignacioct, I think I only changed the name for the property, works for you?:

```python
@property
def num_threads(self) -> Union[int, None]:
    if self.thread_pool_executor:
        return self.thread_pool_executor._max_workers

def __repr__(self) -> str:
    return f"{self.__class__.__name__}(task={self.task.__class__.__name__}, num_threads={self.num_threads}, promp_format='{self.prompt_format}', model='{self.model_name}')"

def __rich_repr__(self) -> Generator[Any, None, None]:
    yield "task", self.task
    yield "num_threads", self.num_threads
    yield "prompt_format", self.prompt_format
    if self.prompt_formatting_fn is not None:
        args = f"({', '.join(self.prompt_formatting_fn.__code__.co_varnames)})"
        representation = self.prompt_formatting_fn.__name__ + args
        yield "prompt_formatting_fn", representation
    yield "model", self.model_name
```

From the quick example :
```python
>>> generator
OpenAILLM(task=TextGenerationTask, num_threads=None, promp_format='None', model='gpt-3.5-turbo')
>>> from rich import print as rprint
>>> rprint(generator)
OpenAILLM(
    task=TextGenerationTask(
        system_prompt="You are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe. Your answers should not include any harmful, unethical, racist,
sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.\nIf a question does not make any sense, or is not factually coherent,
explain why instead of answering something not correct. If you don't know the answer to a question, please don't share false information.",
        task_description=None,
        input_args_names=['input'],
        output_args_names=['generations']
    ),
    num_threads=None,
    prompt_format=None,
    model='gpt-3.5-turbo',
    parameters={'max_tokens': 512, 'frequency_penalty': 0.0, 'presence_penalty': 0.0, 'temperature': 1.0, 'top_p': 1.0}
)
```

Reply:
@plaguss thanks! now it works. I dunno exactly why my name wasn't working, buy nevertheless now it's functional.